Sources/DistributedActors/Plugins/ClusterSingleton/ClusterSingleton.swift

```diff
@@ -109,7 +110,8 @@ internal distributed actor ClusterSingletonBoss<Act: DistributedActor>: _Cluster
 
     private func updateTargetNode(node: UniqueNode?) async throws {
         guard self.targetNode != node else {
-            self.log.debug("Skip updating target node. New node is already the same as current targetNode.", metadata: self.metadata())
+            let metadata = await self.metadata()
```

Comment:
oh wow this is bad, definitely not async logger metadata access. (Why it's bad: this will always render all the metadata, AND causes suspensions even if we're not even logging -- the logging must be done inside the `debug(...)` call because it is lazily evaluated (metadata is an auto closure)

It's because the move of the singleton in to the other actor -- `if let singleton = await self.singleton {`
but why did we do this in a new actor to begin with...? This actor should be able to do everything necessary?

Reply:
amended https://github.com/apple/swift-distributed-actors/pull/980/commits/3b882d62eb53c543108e291f2ffdab95105224de